test/lit/passes/type-ssa.wast

```diff
@@ -0,0 +1,221 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: foreach %s %t wasm-opt           --type-ssa -all -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt --nominal --type-ssa -all -S -o - | filecheck %s --check-prefix NOMNL
+
+;; Test in both isorecursive and nominal modes to make sure we create the new
+;; types properly in both.
+
+;; Every struct.new here should get a new type.
+(module
+  ;; CHECK:      (type $struct (struct (field i32)))
+  ;; NOMNL:      (type $struct (struct (field i32)))
+  (type $struct (struct_subtype (field i32) array))
```

Comment:
```suggestion
  (type $struct (struct_subtype (field i32) data))
```

This still needs to be "data", since it's not a real type but rather placeholder syntax from before we had an official text format for supertypes. I'll fix it to accept "array" and "struct" though.

Reply:
Oh, actually, a better solution would be to not use `struct_subtype` here at all. It can just be a normal `struct`.

Reply:
Meanwhile I flipped it to `data`. Isn't a normal `struct` the old equirecursive notation? I thought `struct_subtype` was the modern/correct form...

Reply:
No, `struct_subtype` is a nonstandard format that we invented to experiment with nominal types. At some point we'll have to port everything over to the standard format and it will be a huge pain :(. Meanwhile you can use plain `(struct ...)` when there is no nontrivial supertype to declare. We should actually prefer that, since that will be valid in the standard format.